zip-0226.rst

```diff
@@ -194,117 +198,144 @@ As in the Orchard protocol, the binding signature verification key, :math:`\math
 
-**Correct Value Commitment:** the following constraints must be added to ensure that the value commitment is computed using the witnessed Asset Base, as represented in the notes:
+Value Commitment Correctness
+''''''''''''''''''''''''''''
+
+The following constraints must be added to ensure that the value commitment is computed using the witnessed Asset Base, as represented in the notes:
 
 - The fixed-base multiplication constraints between the value and the value base point of the value commitment, :math:`\mathsf{cv}`, is replaced with a variable-base multiplication between the two.
-- The witness to the value base point, as defined in the `value base`_ equation is the auxiliary input :math:`\mathsf{AssetBase}^{\mathsf{Orchard}}_{\mathsf{AssetId}}`.
+- The witness to the value base point (as defined in the `asset base`_ equation) is the auxiliary input :math:`\mathsf{AssetBase}^{\mathsf{Orchard}}_{\mathsf{AssetId}}`.
+
+Asset Identifier Consistency for Split Actions
+''''''''''''''''''''''''''''''''''''''''''''''
 
-**Enforce Secure Identifier for Split Actions:** the following constraints must be added to prevent senders from changing the Asset Base for the output note in a Split Action:
+The following constraints must be added to prevent senders from changing the Asset Base for the output note in a Split Action:
 
 - The Value Commitment Integrity should be changed
     - Replace the input note value by a generic value, :math:`\mathsf{v}'`, as :math:`\mathsf{cv^{net}} = \mathsf{ValueCommit_rcv^{OrchardZSA}(v’ - v^new, \mathsf{AssetBase}^{\mathsf{Orchard}}_{\mathsf{AssetId}})}`
-- Add a boolean ``split`` variable as an auxiliary witness. This variable is to be activated ``split = 1`` if the Action in question is a split and ``split = 0`` if the Action is actually spending an input note:
-    - If ``split = 1`` then set :math:`\mathsf{v}' = 0` otherwise :math:`\mathsf{v}'=\mathsf{v^{old}}` from the auxiliary input.
+- Add a boolean ``split_flag`` variable as an auxiliary witness. This variable is to be activated ``split_flag = 1`` if the Action in question has a Split Input and ``split_flag = 0`` if the Action is actually spending an input note:
+    - If ``split_flag = 1`` then set :math:`\mathsf{v}' = 0` otherwise :math:`\mathsf{v}'=\mathsf{v^{old}}` from the auxiliary input.
 - The Merkle Path Validity should check the existence of the note commitment as usual (and not like with dummy notes):
-    - Check that (path, pos) is a valid Merkle path of depth :math:`\mathsf{MerkleDepth^Orchard}`, from :math:`\mathsf{cm^old}` to the anchor :math:`\mathsf{rt^Orchard}`.
+    - Check that (path, pos) is a valid Merkle path of depth :math:`\mathsf{MerkleDepth^Orchard}`, from :math:`\mathsf{cm^{old}}` to the anchor :math:`\mathsf{rt^{Orchard}}`.
```

Comment:
I think it needs to be: "If split_note = 1 or $\mathsf{v^{old}} \neq 0$ then check that (path, pos) ..."

Reply:
This suggestion was incorrect. When is_native_asset $= 0$, we need to force $\mathsf{root} = \mathsf{anchor}$ regardless of split_flag. With my original suggestion it would be possible to use a zero-valued ZSA note to violate balance, because the adversary could set $\mathsf{v^{old}} = 0$, $\mathsf{v^{new}} \neq 0$, is_native_asset $= 0$, split_flag $= 0$, and use an arbitrary asset base for the new note.

Reply:
I've checked that this is fixed.